### DISCUSSION TITLE
Add support OauthTokenExchangeHandler metadata

### DISCUSSION THREAD

**Discussion by mkreth** [forcedotcom/cli#2736] [Ideas]
The Spring '24 release introduced a new OAuth authentication flow - the Token Exchange Flow: https://help.salesforce.com/s/articleView?id=release-notes.rn_security_token_exchange.htm&release=248&type=5

Setting up the Token Exchange Flow requires an Apex class, configuring a Connected App and defining a new Metadata OauthTokenExchangeHandler (see https://help.salesforce.com/s/articleView?id=sf.remoteaccess_token_exchange_handler_define.htm&type=5). When trying to retrieve or deploy (in source format) this new metadata, the CLI errors out with "Missing metadata type definition in registry for id 'oauthtokenexchangehandler'".

It would be great to have support for this metadata.

**Comment by jshackell-sfdc** [comment]
This issue is fixed in CLI version 2.30.8.

**Reply by jaapvandermolen** [reply]
Still getting this error in CLI version 2.34.7

**Reply by cristiand391** [reply]
@jaapvandermolen please open a new issue and follow the bug report template:
https://github.com/forcedotcom/cli/issues/new/choose
